crates/cxx-qt-lib/build.rs

```diff
@@ -4,11 +4,64 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
 use cxx_qt_build::CxxQtBuilder;
+use std::path::PathBuf;
+
+fn qt_gui_enabled() -> bool {
+    std::env::var("CARGO_FEATURE_QT_GUI").is_ok()
+}
+
+fn qt_qml_enabled() -> bool {
+    std::env::var("CARGO_FEATURE_QT_QML").is_ok()
+}
+
+fn qt_quickcontrols_enabled() -> bool {
+    std::env::var("CARGO_FEATURE_QT_QUICKCONTROLS").is_ok()
+}
+
+fn header_dir() -> PathBuf {
+    cxx_qt_build::dir::header_root().join("cxx-qt-lib")
+}
+
+fn write_headers_in(subfolder: &str) {
+    println!("cargo::rerun-if-changed=include/{subfolder}");
+
+    for entry in
+        std::fs::read_dir(format!("include/{subfolder}")).expect("Failed to read include directory")
+    {
+        let entry = entry.expect("Failed to read header file!");
+        let header_name = entry.file_name();
+        println!("cargo::rerun-if-changed=include/{subfolder}/{header_name:?}",);
+
+        // TODO: Do we want to add the headers into a subdirectory?
```

Comment:
btw i was thinking before 1.0 of putting things into `cxx-qt-lib/core/` / ``cxx-qt-lib/gui/` or something like Qt does but unsure.

There is also the debate of whether they should need to include every header of if there could be a single include for all of the headers or per module?

Reply:
That probably makes sense to put it into directories.
We can always add forwarding headers later if we want to add convenience includes.

And we can always add headers that just include all headers of one module, e.g. `core/prelude.h`.

Reply:
I'd say let's do this in a separate PR :)